etl/steps/data/meadow/AI/2023-06-07/monmoth_poll.py

```diff
@@ -0,0 +1,218 @@
+"""Load a snapshot and create a meadow dataset."""
+
+import re
+from typing import cast
+
+import numpy as np
+import pandas as pd
+import pdfplumber
+from owid.catalog import Table
+from structlog import get_logger
+
+from etl.helpers import PathFinder, create_dataset
+from etl.snapshot import Snapshot
+
+# Initialize logger.
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("monmoth_poll.start")
+
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap = cast(Snapshot, paths.load_dependency("monmoth_poll.pdf"))
+
+    # Load data from snapshot.
+    start_page = 3  # tables of interest start on page 3
+    end_page = 6  # and end on page 6
+    texts, dataframes = extract_text_and_tables_from_pdf(snap.path, start_page, end_page)
+
+    # Remove unnecessary strings after extracting survey questions from the pdf (e.g., numbers)
+    texts = remove_strings(texts)
+
+    preprocessed_dataframes = preprocess_data(dataframes, texts)
+
+    # Combine all dataframes into a single dataframe
+    combined_df = pd.concat(preprocessed_dataframes, ignore_index=True)
+
+    # Sort the dataframe by the "year" column
+    # Reset the index
+    combined_df.reset_index(drop=True, inplace=True)
+    # pivot to have
+    pivot_df = combined_df.pivot_table(index=["answer", "year"], columns="question", values="percentage").reset_index()
```

Comment:
I'd recommend always using `pivot` instead of `pivot_table`, since the latter can do unexpected `mean` aggregates under the hood.